chalk-solve/src/clauses/builtin_traits.rs

```diff
@@ -73,6 +77,13 @@ pub fn add_builtin_assoc_program_clauses<I: Interner>(
         WellKnownTrait::DiscriminantKind => {
             discriminant_kind::add_discriminant_clauses(db, builder, self_ty)
         }
+        WellKnownTrait::Generator => {
+            let generalized = generalize::Generalize::apply(db.interner(), self_ty);
```

Comment:
Why do we need this? `trait_ref` (from which `self_ty` is from) is already generalized?

Reply:
We don't have `trait_ref` here (it's in `add_builtin_program_clauses` but not `add_builtin_assoc_program_clauses` and I don't think we can pass it in, unless I'm missing something)

Reply:
Oh, whoops.